core/src/main/scala/cats/data/NonEmptyList.scala

```diff
@@ -133,6 +139,37 @@ final case class NonEmptyList[+A](head: A, tail: List[A]) extends NonEmptyCollec
   def prepend[AA >: A](a: AA): NonEmptyList[AA] =
     NonEmptyList(a, head :: tail)
 
+  /**
+   * Alias for prependAll
+   * 
+   * {{{
+   * scala> import cats.data.NonEmptyList
+   * scala> val nel = NonEmptyList.of(1, 2, 3)
+   * scala> val list = List(-1, 0)
+   * scala> list ++: nel
+   * res0: cats.data.NonEmptyList[Int] = NonEmptyList(-1, 0, 1, 2, 3)
+   * }}}
+   */
+  def ++:[AA >: A](other: List[AA]): NonEmptyList[AA] =
+    prependAll(other)
+
+  /** 
+   * Prepend another List
+   *
+   * {{{
+   * scala> import cats.data.NonEmptyList
+   * scala> val nel = NonEmptyList.of(1, 2, 3)
+   * scala> val list = List(-1, 0)
+   * scala> list ++: nel
+   * res0: cats.data.NonEmptyList[Int] = NonEmptyList(-1, 0, 1, 2, 3)
+   * }}}
+   */
+  def prependAll[AA >: A](other: List[AA]): NonEmptyList[AA] =
+    other match {
+      case Nil          => this
+      case head :: tail => head :: prependAll[AA](tail)
```

Comment:
this is not stack-safe (this is not tail recursive).

I would rather us do: 
```scala
  other match {
    case Nil => this
    case h :: t => NonEmptyList(h, t ::: toList)
  }
```

Reply:
Thanks! I changed it to your suggestions in https://github.com/typelevel/cats/pull/4267/commits/8698f151fe0ab16b9463e69b8b828d9792272075.